### PR TITLE
Use headers only for api and app keys

### PR DESCRIPTION
**Pull request by ssc3**
Datadog now supports api and app keys in headers. Refactoring ruby lib to use headers only

### CODE REVIEW

File: lib/dogapi/common.rb [hunk 1]
```diff
@@ -116,8 +116,10 @@ def request(method, url, extra_params, body, send_json, with_app_key=true)
       resp = nil
       connect do |conn|
         begin
-          current_url = url + prepare_params(extra_params, with_app_key)
+          current_url = url + prepare_params(extra_params)
           req = method.new(current_url)
+          req['DD-API-KEY'] = @api_key
+          req['DD-APPLICATION_KEY'] = @application_key if with_app_key
```
Comment (bkabrda) [inline lib/dogapi/common.rb]: The underscore seems to be problematic here - no requests work with it. If I replace it with a hyphen, so `DD-APPLICATION-KEY`, everything seems to work fine.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Changes
```
[the remark this answers is not in the document]
Comment (ssc3) [reply CHANGELOG.md]: Done

File: lib/dogapi/common.rb (hunk 1 above)
Comment (ssc3) [reply lib/dogapi/common.rb]: Done. Thanks!

**Comment by ssc3** [comment]
@bkabrda I've incorporated your suggestions and added a unit test to check that api and app keys are actually present. Please do continue your review.

Thanks

**Pull request merged**